### PR TITLE
SUBMODULE: added subpath in kubernetes secret

### PR DESCRIPTION
**Pull request by shivani170**
# Description

We do not have an option to set subpath if we want to mount external secret or configmap created using kubectl CLI. Without that we can't mount a file on a directory which already has some existing files. It removes the existing files in the directory

**To Reproduce**

1. Create a secret/configmap using CLI
2. Mount it to a directory which already has some files

**Expected behavior**

File from the secret/configmap should be mounted and existing files should also be there

**Current behavior** 

It removes all the existing files and only the file from secret/configmap is there

Fixes https://github.com/devtron-labs/devtron/issues/1385

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/secrets/Secret.tsx [hunk 1]
```diff
@@ -54,6 +54,7 @@ const Secret = ({ respondOnSuccess, ...props }) => {
         try {
             const appChartRefRes = await getAppChartRef(appId);
             const { result } = await getSecretList(appId);
+        console.log(result)
```
Comment (vivek-devtron) [inline src/components/secrets/Secret.tsx:57]: Please remove console.log

**Review by sohel-rp — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

File: src/components/configMaps/ConfigMap.tsx [hunk 2]
```diff
@@ -475,15 +491,27 @@ export function ConfigMapForm({ appChartRef, id, appId, name = "", external, dat
                                 subPath
                             </a>
                             for volume mount)<br></br>
-                            {isSubPathChecked ? <span className="mb-0 cn-5 fs-11">Keys will be used as filename for subpath</span> : null}
+                            {isSubPathChecked ? <span className="mb-0 cn-5 fs-11">{ isExternalValues ? 'Please provide keys of config map to be mounted' : 'Keys will be used as filename for subpath'}</span> : null}
+
```
Comment (vivek-devtron) [inline src/components/configMaps/ConfigMap.tsx:495]: Can we change "isSubPathChecked ?" to "isSubPathChecked  &&"?

File: src/components/configMaps/ConfigMap.tsx [hunk 3]
```diff
@@ -475,15 +491,27 @@ export function ConfigMapForm({ appChartRef, id, appId, name = "", external, dat
...
+                            {isSubPathChecked ? <span className="mb-0 cn-5 fs-11">{ isExternalValues ? 'Please provide keys of config map to be mounted' : 'Keys will be used as filename for subpath'}</span> : null}
+                           
                             {isChartVersion309OrBelow ? <span className="fs-12 fw-5">
                                 <span className="cr-5">Supported for Chart Versions 3.10 and above.</span>
                                 <span className="cn-7 ml-5">Learn more about </span>
                                 <a href="https://docs.devtron.ai/user-guide/creating-application/deployment-template" rel="noreferrer noopener" target="_blank">Deployment Template &gt; Chart Version</a>
                             </span> : null}
                         </span>
                     </Checkbox>
+                    { isExternalValues  && isSubPathChecked? <div className="mb-16">
```
Comment (sohel-rp) [inline src/components/configMaps/ConfigMap.tsx:503]: same here

File: src/components/secrets/Secret.tsx [hunk 4]
```diff
@@ -586,6 +604,16 @@ export const SecretForm: React.FC<SecretFormProps> = function (props) {
                     </span>
                 </Checkbox>
             </div> : ""}
+            {selectedTab === 'Data Volume' && !isExternalValues && isSubPathChecked ? <div className="mb-16">
```
Comment (sohel-rp) [inline src/components/secrets/Secret.tsx:607]: same here

**Review by sohel-rp — APPROVED** [review]
Recent changes related to white space validation & ternary replacement look good 👍

**Pull request merged**